File: junqo_back/src/company-profiles/dto/company-profile.dto.ts
import {
  IsString,
  MinLength,
  MaxLength,
  IsUUID,
  IsNotEmpty,
  IsOptional,
  IsUrl,
} from 'class-validator';
import {
  MAX_NAME_LENGTH,
  MIN_NAME_LENGTH,
} from '../../shared/user-validation-constants';
import { Expose } from 'class-transformer';
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';

// Company Profile retrieved from database
export class CompanyProfileDTO {
  @ApiProperty({
    description: 'User ID that owns this profile',
    format: 'uuid',
    example: '123e4567-e89b-12d3-a456-426614174000',
  })
  @Expose()
  @IsNotEmpty({ message: 'User ID is required' })
  @IsUUID('4', { message: 'User ID must be a valid UUID' })
  userId: string;

  @ApiProperty({
    description: 'Company name',
    example: 'Acme Corporation',
    minLength: MIN_NAME_LENGTH,
    maxLength: MAX_NAME_LENGTH,
  })
  @Expose()
  @IsNotEmpty({ message: 'Name is required' })
  @IsString({ message: 'Name must be a string' })
  @MinLength(MIN_NAME_LENGTH, {
    message: `Name must be at least ${MIN_NAME_LENGTH} characters long`,
  })
  @MaxLength(MAX_NAME_LENGTH, {
    message: `Name must be at most ${MAX_NAME_LENGTH} characters long`,
  })
  name: string;

  @ApiPropertyOptional({
    description: 'URL to company logo image',
    example: 'https://example.com/company-logo.jpg',
  })
  @Expose()
  @IsOptional()
  @IsUrl()
  avatar?: string;

  // Obligatory for use with casl ability
  constructor(data: Partial<CompanyProfileDTO>) {
    Object.assign(this, data);
  }
}

// Expected values to create a Company Profile
export class CreateCompanyProfileDTO extends CompanyProfileDTO {}

// Expected values to update a Company Profile
export class UpdateCompanyProfileDTO {
  @ApiPropertyOptional({
    description: 'URL to company logo image',
    example: 'https://example.com/company-logo.jpg',
  })
  @Expose()
  @IsOptional()
  @IsUrl()
  avatar?: string;
}

File: junqo_old_front/lib/shared/dto/company_profile_dto.dart
class CompanyProfileDTO {
  final String userId;
  final String name;
  final String? avatar; // Optional
  final String? description; // Optional
  final String? websiteUrl; // Optional

  CompanyProfileDTO({
    required this.userId,
    required this.name,
    this.avatar,
    this.description,
    this.websiteUrl,
  });

  factory CompanyProfileDTO.fromJson(Map<String, dynamic> json) {
    return CompanyProfileDTO(
      userId: json['userId'] as String,
      name: json['name'] as String,
      avatar: json['avatar'] as String?,
      description: json['description'] as String?,
      websiteUrl: json['websiteUrl'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = {
      'userId': userId,
      'name': name,
    };
    if (avatar != null) {
      data['avatar'] = avatar;
    }
    if (description != null) {
      data['description'] = description;
    }
    if (websiteUrl != null) {
      data['websiteUrl'] = websiteUrl;
    }
    return data;
  }
} 